compiler/rustc_codegen_llvm/src/abi.rs

```diff
@@ -69,7 +69,9 @@ impl ArgAttributesExt for ArgAttributes {
                 } else {
                     llvm::LLVMRustAddDereferenceableOrNullAttr(llfn, idx.as_uint(), deref);
                 }
+                // dereferenceable implies nonnull noundef; dereferenceable_or_null implies noundef
```

Comment:
While it does "imply" noundef, LLVM doesn't consistently make that implication, see https://llvm.godbolt.org/z/47rsMd1YK. The preference is to make all attributes explicit, not depend on implication.